src/umessage/umessagebuilder.rs

```diff
@@ -713,19 +673,39 @@ impl UMessageBuilder {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn build_with_protobuf_payload<T: Message>(
+    pub fn build_with_protobuf_payload<T: MessageFull>(
         &mut self,
         payload: &T,
-    ) -> Result<UMessage, UMessageBuilderError> {
-        payload
-            .write_to_bytes()
-            .map_err(UMessageBuilderError::from)
-            .and_then(|serialized_payload| {
-                self.build_with_payload(
-                    serialized_payload.into(),
-                    UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF,
-                )
-            })
+    ) -> Result<UMessage, UMessageError> {
+        if !(self.payload_format == UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF
```

Comment:
where is `self.payload_format` set? Before your change, the payload format always was `UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF`. If we now want to also support `UPayloadFormat::UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY`, then we either need to pass in the payload format as a param, add a dedicated function for building with wrapped payload or detect if the passed in payload actually is an `Any`. FMPOV we should go with option 2, making the whole thing very explicit and also making it very easy to later deprecate and remove support for it again ...